sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -2098,6 +2098,192 @@ public virtual async Task<AnalyzeHealthcareEntitiesOperation> StartAnalyzeHealth
 
         #endregion
 
+        #region Dynamic Classify
+
+        /// <summary>
+        /// Performs dynamic classification on the given document.
+        /// For document length limits, maximum batch size, and supported text encoding, see
+        /// <see href="https://aka.ms/azsdk/textanalytics/data-limits"/>.
+        /// </summary>
+        /// <remarks>
+        /// This method is only available for <see cref="TextAnalyticsClientOptions.ServiceVersion.V2022_10_01_Preview"/>, and newer.
+        /// </remarks>
+        /// <param name="document">The document to analyze.</param>
+        /// <param name="language">The language that the documents are written in.</param>
+        /// <param name="options">The additional <see cref="DynamicClassifyOptions"/> used to configure the operation.</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> controlling the lifetime of the request.</param>
+        /// <returns>
+        /// The collection of categories used to classify each document that was successfully analyzed.
+        /// </returns>
+        /// <exception cref="NotSupportedException">This method is only supported in service API version 2022-10-01-preview and newer.</exception>
+        /// <exception cref="RequestFailedException">The service returned a non-success status code.</exception>
+        /// <exception cref="ArgumentException"><paramref name="document"/> is an empty string.</exception>
+        /// <exception cref="ArgumentNullException">
+        /// <para><paramref name="document"/> is null.</para>
+        /// <para><paramref name="options"/> is null.</para>
+        /// </exception>
+        public virtual Response<ClassificationCategoryCollection> DynamicClassify(
+            string document,
+            DynamicClassifyOptions options,
```

Comment:
@heaths, @kinelski: I wanted to ask you what you think about this one: Right now, `DynamicClassifyOptions` has the required `Categories` property, which is just a list of strings to specify which categories participate in the classification. Here's their example, where they use the categories "Health", "Politics", "Music", and "Sports":
🔗 https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/examples/text/SuccessfuDynamicClassificationRequest.json

From the Guidelines: "The Options class is designed similarly to .NET custom attributes, where required service method parameters are modeled as Options class constructor parameters and get-only properties, and optional parameters are get-set properties." 

(I just realized I need to make `Categories` get-only if I wanted to follow this accurately. But before I do that...)

I was wondering if in this case it makes more sense to move the `Categories` property out of the Options class and make it a parameter that must be passed in all these methods (kind of like `document`/`documents` is its own parameter).

Reply:
Depends if it might grow. I'd talk with @tg-msft or @KrzysztofCwalina about this. There's some nuance to this. I've linked the guidelines above as well but it's better to get a final say from them.

Reply:
We're going with the option of moving the `Categories` property out of the `DynamicClassifyOptions` class and making it a parameter. I think it looks better now.